### ISSUE TITLE
[BUG] Tag capability:global_forecasting not added for regression based forecasting

### ISSUE THREAD

**Issue by kdekker-kdr4** [sktime/sktime#7332]
**Describe the bug**
The tag capability:global_forecasting is not added for regression based forecasters.
For example, RecursiveTabularRegressionForecaster does not have the tag, while global forecasting is possible via the option `pooling="global"`.

**To Reproduce**

The code below gives an error that the tag does not exist.

```python
from sktime.forecasting.compose import make_reduction
from sklearn.ensemble import GradientBoostingRegressor
regressor = GradientBoostingRegressor()
forecaster = make_reduction(regressor, window_length=15, strategy="recursive", pooling="global")  # RecursiveTabularRegressionForecaster 
forecaster.get_tag("capability:global_forecasting")
```

**Expected behavior**
The expected behavior is that for any regression based forecasting model the tag "capability:global_forecasting" should return `True`.

**Versions**
0.34.0

**Comment by fkiraly** [comment]
@kdekker-kdr4, the issue is subtle - I think you are correct, *and* the tags are currently set correctly.

The issue is related to the overload of terminology what "global" currently means, see this issue on the problem: https://github.com/sktime/sktime/issues/6690

For the reduction forecaster, indeed the pooling can be set to `global`, but forecasts are not possible on time series that are not the time series seen in `fit` - which is what the tag signifies.

I agree this is counterintuitive, that was part of the motivation behind opening #6690 to get to a resolution (which we have not reached yet).

I would very much appreciate your contribution to issue #6690, perhaps you have a good idea how to resolve this, and the issue with the tag.

On a separate note, the reduction forecasters *could* be extended to the more general "cross-learning" type global forecasting, that should perhaps be an issue. FYI @XinyuWuu, @benHeid

**Comment by kdekker-kdr4** [comment]
Yes my expectation was also that the regression based forecasting could support cross-learning via the global pooling. It would be appreciated if that is made possible.

**Comment by kdekker-kdr4** [comment]
I will add a comment in the #6690 on my perspective.

**Comment by fkiraly** [comment]
> Yes my expectation was also that the regression based forecasting could support cross-learning via the global pooling. It would be appreciated if that is made possible.

Yes, that would be great! Though strictly speaking that is a feature request and not a false tag, even if there is currently a terminology issue.

**Comment by kdekker-kdr4** [comment]
Okay fair.
